### ISSUE TITLE
Error running in Linux: GLIBC not found

### ISSUE THREAD

**Issue by Daveyr** [Androxyde/Flashtool#121]
I am running Ubuntu 18.04.2 LTS with 64-bit Java installed. I have successfully unzipped the 0.9.26.0 Linux version of the software from the flashtool.net site.

When running ./FlashTool as root I receive the following error:

```
Used java home : ./x10flasher_lib/linjre64
ldd: x10flasher_lib/linux/lib32/udev/libusbx-1.0.so.0.1.0: No such file or directory
ls: cannot access './x10flasher_lib/swtlin/x86/*jar': No such file or directory
Error: dl failure on line 597
Error: failed /home/ubuntu/Documents/phone/FlashTool/x10flasher_lib/linjre64/lib/server/libjvm.so, because /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /home/ubuntu/Documents/phone/FlashTool/x10flasher_lib/linjre64/lib/server/libjvm.so)
``` 

There is an odd reference here to GLIBC_2.29. What does it mean? I have verified that there is a valid file called `/lib/x86_64-linux-gnu/libm.so.6` but I cannot verify its version.

**Comment by Daveyr** [comment]
I should note that it's not surprising `ls: cannot access './x10flasher_lib/swtlin/x86/*jar'`. This is because the correct folder should be `.../x86_64/jar`. It's not obvious in the FlashTool script where to edit this though.

**Comment by lin0lin0** [comment]
@Daveyer,
Have exact same error as you got at start - just on Lubuntu 18.04.2 LTS.

Before I found your ticket I thought it is about installed glibcc in my system:
```
lino@lu18:~/bin$ ldd --version
ldd (Ubuntu GLIBC 2.27-3ubuntu1) 2.27
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
lino@lu18:~/bin$ 
```
Is your second entry "kind of solution"? If so could you point me which line to change in the script?

Cheers
lin0lin0

**Comment by Daveyr** [comment]
No, my second entry wasn't a solution although I thought it might be
helpful information for someone to help me. The error message said it was
trying to find a file path that simply isn't part of the installation. I
tried to edit the script to point at the correct folder but this did not
work for me.

If you want to try for yourself I used nano to edit the script and searched
for the file path that matched the error message. I changed it to match the
folder structure that the installation has. Let me know if you succeed!

**Comment by Androxyde** [comment]
Hi, if you have a working openjdk 12 on Ubuntu 18.04, use it in place of the bundled one.
It seems the bundled one (taken right from openjdk website) is built against a more recent GLIBC than the one you have in your system

**Comment by skupfer** [comment]
Tried openjdk 13 on Ubuntu 18 ... Same error.

**Comment by testbughub** [comment]
I get a similar error on Debian 10.
```Running as root.
Error: dl failure on line 603
Error: failed /home/lamarca/Downloads/FlashTool/x10flasher_native/jre/lib/server/libjvm.so, because /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /home/lamarca/Downloads/FlashTool/x10flasher_native/jre/lib/server/libjvm.so)

**Comment by stevenengland** [comment]
> Tried openjdk 13 on Ubuntu 18 ... Same error.

Hi, did you try these steps?

1. Install a jre: sudo apt-get install openjdk-11-jre
2. Edit the startup script: nano ./Flashtool
3. Change the line that points to the JAVA Home:
#export JAVA_HOME=$working_dir/x10flasher_native/jre
export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64

After that I am able to launch the app.

**Comment by kubriel** [comment]
can confirm.

>     3. Change the line that points to the JAVA Home:
>        #export JAVA_HOME=$working_dir/x10flasher_native/jre
>        export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64
> 
> 
> After that I am able to launch the app.

**Comment by RuhanSA079** [comment]
Hello, I am facing the same issue, been running Flashtool on Windows OK for a while, but it seems like the Linux variant is more picky?
Not running as root but Sony/SonyEriccson Vendor ID found on your udev rules
if Flashing didn't work well, run flashtool as root
Error: dl failure on line 534
Error: failed /home/ruhan/Downloads/flashtool/FlashTool/x10flasher_native/jre/lib/server/libjvm.so, because /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /home/ruhan/Downloads/flashtool/FlashTool/x10flasher_native/jre/lib/server/libjvm.so)

I have installed OpenJDK 11 JRE as @stevenengland mentioned, but still no dice.
Also exported the JAVA_HOME env variable.
Tried it with Ubuntu 20.04 as well, it then complains about the same, but GLIBC 2.32.

**Comment by comradekingu** [comment]
Remember to also remove the "$working_dir/". :)

If it says
>Error: LinkageError occurred while loading main class gui.Main
	java.lang.UnsupportedClassVersionError: gui/Main has been compiled by a more recent version of the Java Runtime (class file version 59.0), this version of the Java Runtime only recognizes class file versions up to 55.0

You can download v15 from https://jdk.java.net/ put the extracted jdk-15.0.1 folder in the FlashTool folder and use

export JAVA_HOME=$working_dir/jdk-15.0.1

**Comment by boshtannik** [comment]
It helped, i could run This tool.

**Comment by raspatan** [comment]
@comradekingu this answer worked in Ubuntu 20.04, but folder name was jdk-15
